src/module.c

```diff
@@ -12856,20 +12856,37 @@ long long getModuleNumericConfig(ModuleConfig *module_config) {
     return module_config->get_fn.get_numeric(rname, module_config->privdata);
 }
 
+int loadModuleDefaultConfigs(RedisModule *module) {
+    listIter li;
+    listNode *ln;
+    const char *err = NULL;
+    listRewind(module->module_configs, &li);
+    while ((ln = listNext(&li))) {
+        ModuleConfig *module_config = listNodeValue(ln);
+        if (!performModuleConfigSetDefaultFromName(module_config->name, &err)) {
+            serverLog(LL_WARNING, "Issue attempting to set default value of configuration %s : %s", module_config->name, err);
+            return REDISMODULE_ERR;
+        }
+    }
+    module->configs_initialized |= MODULE_CONFIGS_DEFAULTS;
+    return REDISMODULE_OK;
+}
+
 /* This function takes a module and a list of configs stored as sds NAME VALUE pairs.
  * It attempts to call set on each of these configs. */
 int loadModuleConfigs(RedisModule *module) {
     listIter li;
     listNode *ln;
     const char *err = NULL;
     listRewind(module->module_configs, &li);
+    const int set_default_if_missing = !(module->configs_initialized & MODULE_CONFIGS_DEFAULTS);
```

Comment:
@oranagra discussed with @kei-nan - before this PR when calling `loadModuleConfigs` more than once, all values were reverted back to their defaults, and after this PR, they will not (flag is set to `MODULE_CONFIGS_ALL_APPLIED` at the end of this function)

Looks like a good progression - just pointing this out.

Reply:
i'd consider it a bug fix for a non-realistic scenario where the user wrongly uses the API.
considering the outcome of that wrong use, i think chances are that if someone had that issue, they would have noticed it, so i don't expect it to happen.
i'm ok to proceed.